src/com/tictacgo/TicTacGoGameActivity.java

```diff
@@ -0,0 +1,402 @@
+   */
+  private Player turn;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_tic_tac_go_game);
+
+    Intent intent = getIntent();
+
+    turn = (Player) intent.getSerializableExtra(TicTacGoMenuActivity.playerKey);
+
+    // Undo/redo initialization
+    undoHistory = new ArrayList<>();
+    historyIndex = 0;
+
+    height = intent.getIntExtra(TicTacGoMenuActivity.heightKey, 300);
+    board = new Board(turn, height, getBaseContext());
+
+    // Set up the screen
+    ((TextView) findViewById(R.id.gamePlayerOneName))
+        .setText(intent.getStringExtra(TicTacGoMenuActivity.p1NameKey));
+    ((TextView) findViewById(R.id.gamePlayerTwoName))
+        .setText(intent.getStringExtra(TicTacGoMenuActivity.p2NameKey));
+    fl = (FrameLayout) findViewById(R.id.gameBoard);
+
+    // What to do when an empty space is clicked
+    pieceClicked = new View.OnClickListener() {
+      public void onClick(View v) {
+        if (directions != null) //Popup already active
+          directions.dismiss(); //Dismiss old popup
+        board.makePiece(((FrameLayout.LayoutParams) v.getLayoutParams()).gravity); //Set the location for the new Piece
+        LayoutInflater inflater = getLayoutInflater();
+        View grid = inflater.inflate(R.layout.direction, null); //Show popup
+        directions = new PopupWindow(grid, -2, -2); //Show  popup
+        /**
+         * Offset is used so that the correct edge of the popup window will
+         * line up with the edge of the Board
+         */
+        int xOffset = 0, yOffset = 0; //No offset
+        switch (((FrameLayout.LayoutParams) v.getLayoutParams()).gravity) {
+          case Gravity.TOP | Gravity.LEFT:
+            break;
+          case Gravity.TOP | Gravity.CENTER_HORIZONTAL:
+            xOffset = height / 4;
+            break;
+          case Gravity.TOP | Gravity.RIGHT:
+            xOffset = height / 2;
+            break;
+          case Gravity.CENTER_VERTICAL | Gravity.LEFT:
+            yOffset = height / 4;
+            break;
+          case Gravity.CENTER_VERTICAL | Gravity.RIGHT:
+            xOffset = height / 2;
+            yOffset = height / 4;
+            break;
+          case Gravity.BOTTOM | Gravity.LEFT:
+            yOffset = height / 2;
+            break;
+          case Gravity.BOTTOM | Gravity.CENTER_HORIZONTAL:
+            xOffset = height / 4;
+            yOffset = height / 2;
+            break;
+          case Gravity.BOTTOM | Gravity.RIGHT:
+            xOffset = height / 2;
+            yOffset = height / 2;
+            break;
+          default:
+            xOffset = height / 4;
+            yOffset = height / 4;
+        }
+        directions.showAtLocation(fl, Gravity.NO_GRAVITY, xOffset, yOffset); //Make popup visible
+        int id = R.drawable.pieceodirection; //Which picture to use for directions (O)
```

Comment:
Not sure if this has been taken care of in another PR already, but resource files should be named in snake style (piece_o_direction). That can be in another PR though.